src/frequenz/sdk/microgrid/_graph.py

```diff
@@ -688,6 +711,42 @@ def is_chp_chain(self, component: Component) -> bool:
         """
         return self.is_chp(component) or self.is_chp_meter(component)
 
+    def dfs(
```

Comment:
Sorry for coming late to the party, but wouldn't it make more sense to call this just `search()`? Or `depth_first_search()` if it is very important to note that uses DFS. The name is just too cryptic IMHO if you are not familiar with graphs, I would also make `current_node` and `visited` optional, default them to the root node and an empty set when they are `None`.

Or maybe we can keep `dfs()` as it is and add another higher-level method `search` that does what I say above...